### PR TITLE
Serialize rockconfig

### PR DESCRIPTION
**Pull request by joakim-hove**
Downstream of: https://github.com/OPM/opm-common/pull/1433

### CODE REVIEW

**Comment by joakim-hove** [comment]
Mergeable?

File: opm/simulators/utils/ParallelRestart.hpp [hunk 1]
```diff
@@ -76,7 +76,7 @@ namespace Action {
 
 class Aqudims;
 class BCConfig;
-class BCConfig::BCFace;
+struct BCConfig::BCFace;
```
Comment (akva2) [inline opm/simulators/utils/ParallelRestart.hpp]: this is useless. 
- you are prototyping an existing structure (ie already included)
- you can never prototype a member of another class like this in any case.

Comment (akva2) [reply opm/simulators/utils/ParallelRestart.hpp]: (I already remove this in https://github.com/OPM/opm-simulators/pull/2305 so my preferred fix is for you not to touch it, so i avoid the rebase)

Comment (joakim-hove) [reply opm/simulators/utils/ParallelRestart.hpp]: OK - done.

**Pull request merged**